File: Eatable2023Data/Product/Store.cs
using Eatable.Common.Enums;
using Eatable.Data.General;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Eatable.Data.Product
{
    public class Store: Base
    {
        [Key]
        public Guid StoreId { get; set; }

        [Required]
        [MaxLength(50)]
        public string StoreIdentifier { get; set; }

        [Required]
        [MaxLength(100)]
        public string StoreName { get; set; }

        public Address Address { get; set; }

        public StoreType StoreType { get; set; }

        public string OpeningHours { get; set; }

        public List<ContactInformation> ContactInformation { get; set; }

        [MaxLength(100)]
        public string StoreUrl { get; set; }

        [MaxLength(100)]
        public string LogoName { get; set; }
    }
}

File: EatableAngTest/EatableTest/src/app/models/store.ts
import { IAddress } from "./address";
import { IContactInformation } from "./contactInformation";
import { StoreType } from "./enums/storeType";

export interface IStore {

    storeId: string;
    storeIdentifier: string;
    storeName: string;
    address?: null | IAddress;
    storeType: StoreType;
    openingHours?: null | string;
    contactInformation?: null | Array<IContactInformation>;
  }